manager/allocator/networkallocator/networkallocator.go

```diff
@@ -186,12 +187,28 @@ func (na *NetworkAllocator) ServiceAllocate(s *api.Service) (err error) {
 		return
 	}
 
-	// First allocate VIPs for all the pre-populated endpoint attachments
-	for _, eAttach := range s.Endpoint.VirtualIPs {
-		if err = na.allocateVIP(eAttach); err != nil {
-			return
+	// Allocate VIPs for all the pre-populated endpoint attachments
```

Comment:
Also unrelated to the changes here, but I think it would be a great cleanup to split the contents of this function into `serviceAllocateVIP` and `serviceAllocateDNSRR` functions.